git/config.py

```diff
@@ -419,6 +419,8 @@ def string_decode(v):
                         optval = string_decode(optval[1:])
                     # end handle multi-line
                     # preserves multiple values for duplicate optnames
+                    # fix for extra back slashes so windows UNC paths are not mangled
+                    optval = optval.replace('\\\\', '\\')
```

Comment:
Have you checked what happens if that file is written back to disk?
Maybe this change could be constrained to files that are open for reading only (if this is even known).

Otherwise I fear that writing such a mangled file back, for example because some other value changed, will cause it to become invalid for some users.

What is your thoughts on this?

Reply:
It looks like it might be platform dependent. Config entries get written [here](https://github.com/gitpython-developers/GitPython/blob/30387f16920f69544fcc7db40dfae554bcd7d1cc/git/config.py#L516) and the `defenc` encoding is determined [here](https://github.com/gitpython-developers/GitPython/blob/30387f16920f69544fcc7db40dfae554bcd7d1cc/git/compat.py#L24). I'll test it on my win10 machine, and if it works, we might wrap that `replace` call in an [`if is_win:`](https://github.com/gitpython-developers/GitPython/blob/30387f16920f69544fcc7db40dfae554bcd7d1cc/git/compat.py#L21) statement as a precaution.